### PR TITLE
Carl ansi part1

### PR DESCRIPTION
**Pull request by ccodere**
Let us try again : This is part 1 of the ANSI conversion, i could not do it in a smaller chunk than this.

Next steps: 
* Adapt the lua scripts to use the new sed scripts.
* Continue base ansi C conversion of other compilers.
* Add more testing on C compiler recompilation with different flags.

Later on:
* Get rid of system as much as possible.

Next steps should be smaller than this one.

### CODE REVIEW

**Comment by ccodere** [comment]
I will look into it tomorrow, this was building on my machine, but after the merge with the head and latest changes i get on my local machine also:

File: modules/src/em_code/em.c [hunk 1]
```diff
@@ -69,6 +72,7 @@ C_flush() {
 		return;
 	}
 #endif
+	count = (int)(C_opp - obuf);
```
Comment (davidgiven) [inline modules/src/em_code/em.c]: What's this variable for? Was it supposed to be used in sys_write() below?

File: modules/src/em_data/makeem.c [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+}
+
+
+
+
+/** Copies a null terminated string,
+ *  but allocates the memory.
+ *
+ */
+char* strdup(const char* s)
```
Comment (davidgiven) [inline modules/src/em_data/makeem.c]: strdup() is in <string.h>.

File: modules/src/em_data/makeem.c [hunk 3]
```diff
@@ -0,0 +1,309 @@
+/** A utility to convert the em definition table to generated code.
```
Comment (davidgiven) [inline modules/src/em_data/makeem.c]: I'm not keen on this --- it replaces the existing 77 line script with 300 lines of C, which is very much the wrong language for text processing. OTOH it does remove the ed dependency, which is great. Something like awk or Lua (there's a Lua interpreter already in the build system; see mach/mips/as/build.lua for how to call it) would probably reduce the size by 2/3.


Also, this isn't hooked into the build system, which still uses the shell script.

File: util/LLgen/src/LLgen.c [hunk 4]
```diff
@@ -2,9 +2,6 @@
 #include "Lpars.h"
```
Comment (davidgiven) [inline util/LLgen/src/LLgen.c:2]: Just checking: this entire file is autogenerated by LLgen and you didn't edit it by hand, right?

**Comment by davidgiven** [comment]
Wow, this was a tonne of work! Thanks very much. I've gon through it and left a few minor comments. LGTM in general.

My biggest remaining concern is that it's too big, and is doing three different things: doing the ansification, adding part of a cmake setup, and replacing some of the source generation scripts.

Could you please move the cmake stuff and the script refactoring into different PRs? The cmake stuff is only partially complete and isn't ready for merge. The new scripts aren't hooked into the build.lua build system, so they're not used anyway, which means they shouldn't be here. We want to ansification stuff to be as simple and mechanical as possible.

But thank you for doing all this! It's really useful.

File: modules/src/em_code/em.c (hunk 1 above)
Comment (ccodere) [reply modules/src/em_code/em.c]: Legacy code when debugging. It has been removed.

File: modules/src/em_data/makeem.c (hunk 3 above)
Comment (ccodere) [reply modules/src/em_data/makeem.c]: I have removed this from the branch, I will try to convert it to an AWK script instead.

File: util/LLgen/src/LLgen.c (hunk 4 above)
Comment (ccodere) [reply util/LLgen/src/LLgen.c:2]: Yes, you are correct. I did not touch this part.

**Comment by ccodere** [comment]
> 
> Could you please move the cmake stuff and the script refactoring into different PRs? The cmake >stuff is only partially complete and isn't ready for merge. The new scripts aren't hooked into the >build.lua build system, so they're not used anyway, which means they shouldn't be here. We want to >ansification stuff to be as simple and mechanical as possible.

Done, normally CMake stuff and sed scripts are completely removed, not sure if git automatically removes the scripts directory though.

**Comment by davidgiven** [comment]
Thanks! That's merged --- it looks great. It's a substantially cleaner compile now, too.

**Pull request merged**